### PR TITLE
CM-1513-permissions for users

### PR DESCRIPTION
**Pull request by watchtower314**

### CODE REVIEW

File: packages/firebase/functions/src/permissions/triggers/onCommonCreated.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import {Collections} from '../../constants';
+import {IEventEntity} from '../../event/types';
+import {addPermission} from '../../permissions/business';
+import { EVENT_TYPES } from '../../event/event';
+
+export const onCommonCreated = functions.firestore
+  .document(`/${Collections.Events}/{id}`)
+  .onCreate(async (eventSnap) => {
+    const event = eventSnap.data() as IEventEntity;
+    if (event.type === EVENT_TYPES.COMMON_CREATED) {
```
Comment (watchtower314) [inline packages/firebase/functions/src/permissions/triggers/onCommonCreated.ts:13]: @alexander2001i 10x!!!! It worked!

Comment (alexander2001i) [reply packages/firebase/functions/src/permissions/triggers/onCommonCreated.ts:13]: 👾👾👾

File: packages/firebase/functions/src/permissions/business/addPermission.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+
+/**
+ * Updating permission of a user to make changes in a common
+ * - For creating common, the user (userId) will get a 'founder' permission
+ * - For a user (requestByUserId) asking to set permission to another user (userId),
+ * we check that requestByUserId has sufficient permission to grant permission (getting this requestByUserId from the http request)
+ * and if that user has permission (right now, meaning being founder) then userId will get the permission
+ * 
+ * @return               [description]
```
Comment (alexander2001i) [inline packages/firebase/functions/src/permissions/business/addPermission.ts]: Descriptive 😅

File: packages/firebase/functions/src/permissions/types.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+	/**
+	 * The commonId of the common that a user has permission to
+	 */
+	data?: {
+		commonId?: string
+	}
+}
+
+export type Role = 'founder'| 'moderator' | 'other';
```
Comment (alexander2001i) [inline packages/firebase/functions/src/permissions/types.ts]: @common/types

**Review by alexander2001i — CHANGES_REQUESTED** [review]

File: packages/firebase/functions/src/permissions/business/addPermission.ts (hunk 2 above)
Comment (watchtower314) [reply packages/firebase/functions/src/permissions/business/addPermission.ts]: 😂

**Comment by watchtower314** [comment]
@alexander2001i not quite sure what to do with these 3 eslint errors :)

**Comment by alexander2001i** [comment]
@watchtower314 I fixed the two import related issues, but am not sure why it does not allow the `I`. The lint returns success locally. In any case can you please move the user types to the types package?

File: packages/firebase/functions/src/util/tests/helpers/mockers/getUser.mocker.ts [hunk 4]
```diff
@@ -1,4 +1,5 @@
-import { IUserEntity } from '../../../../users/types';
+/*global firestore jest*/
```
Comment (watchtower314) [inline packages/firebase/functions/src/util/tests/helpers/mockers/getUser.mocker.ts]: Screen Shot 2021-02-10 at 5 29 09

File: packages/firebase/functions/src/notification/notification.ts [hunk 5]
```diff
@@ -11,44 +12,33 @@ import { EVENT_TYPES } from '../event/event';
...
 import { getUserById } from '../util/db/userDbService';
+import { discussionDb } from '../discussion/database';
 import { Utils } from '../util/util';
 import { ISendTemplatedEmailData } from './email';
 import { getFundingRequestAcceptedTemplate } from './helpers';
 
 const messaging = admin.messaging();
 
-const getNameString = (userData) => {
+const getNameString = (userData): string => {
```
Comment (alexander2001i) [inline packages/firebase/functions/src/notification/notification.ts:23]: Nice! Not also type `userData`

**Review by alexander2001i — APPROVED** [review]

**Pull request merged**